tests/bluetooth/mesh/rpl/src/main.c

```diff
@@ -0,0 +1,480 @@
+	skip_delete = true;
+	ztest_expect_value(bt_mesh_settings_store_schedule, flag, BT_MESH_SETTINGS_RPL_PENDING);
+	bt_mesh_rpl_clear();
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+	skip_delete = false;
+
+	settings_func = 0;
+	settings_func_cnt = 0;
+}
+
+/**** Mocked functions ****/
+
+void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
+{
+	ztest_check_expected_value(flag);
+}
+
+void bt_mesh_settings_store_cancel(enum bt_mesh_settings_flag flag)
+{
+}
+
+int settings_save_one(const char *name, const void *value, size_t val_len)
+{
+	ztest_check_expected_data(name, strlen(name));
+
+	check_op(SETTINGS_SAVE_ONE);
+
+	return 0;
+}
+
+int settings_delete(const char *name)
+{
+	if (skip_delete) {
+		return 0;
+	}
+
+	ztest_check_expected_data(name, strlen(name));
+
+	check_op(SETTINGS_DELETE);
+
+	return 0;
+}
+
+/**** Tests ****/
+
+ZTEST_SUITE(bt_mesh_rpl_reset, NULL, NULL, setup, NULL, NULL);
+
+/** Test that entries with old_iv == true are removed after the reset operation finished. */
+ZTEST(bt_mesh_rpl_reset, test_reset_normal)
+{
+	prepare_rpl_and_start_reset();
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		if (test_vector[i].old_iv) {
+			ztest_expect_value(settings_delete, name, test_vector[i].name);
+		} else {
+			ztest_expect_data(settings_save_one, name, test_vector[i].name);
+		}
+	}
+
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+
+	check_entries_from_test_vector();
+	check_empty_entries(EMPTY_ENTRIES_CNT);
+}
+
+/** Test that RPL accepts and stores a valid entry that was just deleted. The entry should be
+ * stored after the reset operation is finished.
+ */
+ZTEST(bt_mesh_rpl_reset, test_rpl_check_on_delete_same_entry)
+{
+	prepare_rpl_and_start_reset();
+	expect_pending_store();
+
+	/* Take the first entry with old_iv == true and simulate msg reception with same src
+	 * address and correct IVI after the entry was deleted.
+	 */
+	struct test_rpl_entry *entry = &test_vector[1];
+
+	zassert_true(entry->old_iv);
+	entry->old_iv = false;
+	call_rpl_check_on(SETTINGS_DELETE, 1, entry);
+
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+	zassert_true(is_rpl_check_called());
+
+	/* Call bt_mesh_rpl_pending_store() to store new entry. */
+	ztest_expect_data(settings_save_one, name, entry->name);
+	bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
+
+	check_entries_from_test_vector();
+	check_empty_entries(EMPTY_ENTRIES_CNT);
+}
+
+/** Test that RPL accepts and stores a valid entry that was just stored. The entry should be stored
+ * after the reset operation is finished.
+ */
+ZTEST(bt_mesh_rpl_reset, test_rpl_check_on_save_same_entry)
+{
+	prepare_rpl_and_start_reset();
+	expect_pending_store();
+
+	/* Take the first entry with old_iv == false and simulate msg reception with same src
+	 * address and correct IVI after the entry was stored.
+	 */
+	struct test_rpl_entry *entry = &test_vector[0];
+
+	zassert_false(entry->old_iv);
+	entry->old_iv = false;
```

Comment:
```suggestion
	zassert_false(entry->old_iv);
```
Is it not redundant to set the old_iv to false after checking that it is already set to false?